module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -135,19 +229,20 @@ public function authenticate($request)
         $catPassword = null;
 
         // Has the user configured attributes to use for populating the user table?
-        $attribsToCheck = [
-            'cat_username', 'cat_password', 'email', 'lastname', 'firstname',
-            'college', 'major', 'home_library'
-        ];
-        foreach ($attribsToCheck as $attribute) {
-            if (isset($shib->$attribute)) {
-                $value = $request->getServer()->get($shib->$attribute);
+        foreach ($this->attribsToCheck as $attribute) {
+            if (isset($shib[$attribute])) {
+                $value = $this->getAttribute($request, $shib[$attribute]);
                 if ($attribute == 'email') {
                     $user->updateEmail($value);
-                } elseif ($attribute != 'cat_password') {
-                    $user->$attribute = ($value === null) ? '' : $value;
-                } else {
+                } elseif ($attribute == 'cat_username' && isset($shib['prefix'])
+                    && !empty($value)
+                ) {
+                    $user->cat_username = $shib['prefix'] . '.'
+                        . (($value === null) ? '' : $value);
```

Comment:
This check is no longer necessary since there's the `!empty($value)` check above.